### PR TITLE
AM-11328: Update the missing parameter description in the OSS docs.

### PR DESCRIPTION
**Pull request by uma-kt**

### CODE REVIEW

**Review by narmidm — APPROVED** [review]

File: versioned_docs/version-1.0.0/install-kubeslice/kubeslice-cli/command-reference.mdx [hunk 1]
```diff
@@ -602,8 +602,8 @@ The following are the `kubeslice-cli install` command options.
...
+| --skip              | -s        | It skips the installation steps (comma-separated) with `minimal-demo`, `full-demo`, and `enterprise-demo` options.                    | **kind**: skips the creation of kind clusters. **calico**: skips the installation of Calico. **controller**: skips the installation of KubeSlice Controller. **worker-registration**: skips the registration of worker clusters with the KubeSlice Controller. **worker**: skips the installation of worker clusters. **demo**: skips the installation of additional example applications. **prometheus**: skips the installation of the Prometheus.                          |
```
Comment (veenamj-avesha) [inline versioned_docs/version-1.0.0/install-kubeslice/kubeslice-cli/command-reference.mdx:606]: Is enterprise-demo valid for the OSS edition? I don't think so... 

Are you just mentioning as an option?

File: versioned_docs/version-1.0.0/install-kubeslice/kubeslice-cli/command-reference.mdx [hunk 2]
```diff
@@ -593,7 +593,7 @@ Use this command to install the required workloads to run KubeSlice Controller a
 ```
 kubeslice-cli --config <path-to-the-custom-topology-file> <install|i>
 kubeslice-cli install <command-options>
-kubeslice-cli install --profile=<minimal-demo|full-demo>
+kubeslice-cli install --profile=<minimal-demo|full-demo|enterprise-demo >
```
Comment (veenamj-avesha) [inline versioned_docs/version-1.0.0/install-kubeslice/kubeslice-cli/command-reference.mdx:596]: Same question as below on `enterprise-demo` .. does it hold good for the OSS edition?

File: versioned_docs/version-1.0.0/install-kubeslice/kubeslice-cli/command-reference.mdx (hunk 1 above)
Comment (veenamj-avesha) [inline versioned_docs/version-1.0.0/install-kubeslice/kubeslice-cli/command-reference.mdx:606]: Same question on the enterprise-demo option

File: versioned_docs/version-1.0.0/install-kubeslice/kubeslice-cli/topology-configuration.mdx [hunk 3]
```diff
@@ -85,6 +87,34 @@ The following table describes the worker chart configuration parameters used to
 | version    | Numeric        | The version of the chart. Leave this value blank for the latest version.         | Optional  |
 | values     | String         | The additional value that can be passed as --set arguments for **helm install**. | Optional  |
 
+## UI Chart Configuration Parameters
```
Comment (veenamj-avesha) [inline versioned_docs/version-1.0.0/install-kubeslice/kubeslice-cli/topology-configuration.mdx:90]: UI chart is not valid for OSS

File: versioned_docs/version-1.0.0/install-kubeslice/kubeslice-cli/topology-configuration.mdx [hunk 4]
```diff
@@ -85,6 +87,34 @@ The following table describes the worker chart configuration parameters used to
...
+The following table describes the UI chart configuration parameters used to create topology YAML.
+
+| Parameter  | Parameter Type | Description                                                                      | Required  |
+| ---------- | -------------- | -------------------------------------------------------------------------------- | --------- |
+| chart_name | String         | The name of the UI or enterprise chart.                                          | Mandatory |
+| version    | Numeric        | The version of the chart. Leave it blank for the latest version.                 | Optional  |
+| values     | String         | The additional value that can be passed as --set arguments for **helm install**. | Optional  |
+
+## Prometheus Chart Configuration parameters
```
Comment (veenamj-avesha) [inline versioned_docs/version-1.0.0/install-kubeslice/kubeslice-cli/topology-configuration.mdx:99]: Are these required for OSS also?

File: versioned_docs/version-1.0.0/install-kubeslice/kubeslice-cli/topology-configuration.mdx [hunk 5]
```diff
@@ -85,6 +87,34 @@ The following table describes the worker chart configuration parameters used to
...
+The following table describes the Prometheus chart configuration parameters used to create topology YAML.
+
+| Parameter  | Parameter Type | Description                                                                      | Required  |
+| ---------- | -------------- | -------------------------------------------------------------------------------- | --------- |
+| chart_name | String         | The name of the Prometheus chart.                                                | Mandatory |
+| version    | Numeric        | The version of the chart. Leave it blank for the latest version.                 | Optional  |
+| values     | String         | The additional value that can be passed as --set arguments for **helm install**. | Optional  |
+
+## Image Pull Secret Configuration Parameters
```
Comment (veenamj-avesha) [inline versioned_docs/version-1.0.0/install-kubeslice/kubeslice-cli/topology-configuration.mdx:108]: These are not required in the OSS edition...

File: versioned_docs/version-1.1.0/install-kubeslice/kubeslice-cli/topology-configuration.mdx [hunk 6]
```diff
@@ -85,6 +87,34 @@ The following table describes the worker chart configuration parameters used to
 | version    | Numeric        | The version of the chart. Leave this value blank for the latest version.         | Optional  |
 | values     | String         | The additional value that can be passed as --set arguments for **helm install**. | Optional  |
 
+## UI Chart Configuration Parameters
```
Comment (veenamj-avesha) [inline versioned_docs/version-1.1.0/install-kubeslice/kubeslice-cli/topology-configuration.mdx:90]: Same comment as above

File: versioned_docs/version-1.1.0/install-kubeslice/kubeslice-cli/topology-configuration.mdx [hunk 7]
```diff
@@ -85,6 +87,34 @@ The following table describes the worker chart configuration parameters used to
...
+The following table describes the UI chart configuration parameters used to create topology YAML.
+
+| Parameter  | Parameter Type | Description                                                                      | Required  |
+| ---------- | -------------- | -------------------------------------------------------------------------------- | --------- |
+| chart_name | String         | The name of the UI or enterprise chart.                                          | Mandatory |
+| version    | Numeric        | The version of the chart. Leave it blank for the latest version.                 | Optional  |
+| values     | String         | The additional value that can be passed as --set arguments for **helm install**. | Optional  |
+
+## Prometheus Chart Configuration parameters
```
Comment (veenamj-avesha) [inline versioned_docs/version-1.1.0/install-kubeslice/kubeslice-cli/topology-configuration.mdx:99]: same comment as above

File: versioned_docs/version-1.0.0/install-kubeslice/kubeslice-cli/command-reference.mdx (hunk 1 above)
Comment (uma-kt) [reply versioned_docs/version-1.0.0/install-kubeslice/kubeslice-cli/command-reference.mdx:606]: kubeslice-cli is a tool, anyone can download and use for OSS or Ent. When the user runs the `help` command he can see all these options.

File: versioned_docs/version-1.1.0/install-kubeslice/kubeslice-cli/topology-configuration.mdx [hunk 8]
```diff
@@ -85,6 +87,34 @@ The following table describes the worker chart configuration parameters used to
...
+The following table describes the Prometheus chart configuration parameters used to create topology YAML.
+
+| Parameter  | Parameter Type | Description                                                                      | Required  |
+| ---------- | -------------- | -------------------------------------------------------------------------------- | --------- |
+| chart_name | String         | The name of the Prometheus chart.                                                | Mandatory |
+| version    | Numeric        | The version of the chart. Leave it blank for the latest version.                 | Optional  |
+| values     | String         | The additional value that can be passed as --set arguments for **helm install**. | Optional  |
+
+## Image Pull Secret Configuration Parameters
```
Comment (veenamj-avesha) [inline versioned_docs/version-1.1.0/install-kubeslice/kubeslice-cli/topology-configuration.mdx:108]: Same comment as above

File: versioned_docs/version-1.0.0/install-kubeslice/kubeslice-cli/topology-configuration.mdx (hunk 3 above)
Comment (uma-kt) [reply versioned_docs/version-1.0.0/install-kubeslice/kubeslice-cli/topology-configuration.mdx:90]: The tool description is kept generic  so user will know the ent options too

File: versioned_docs/version-1.0.0/install-kubeslice/kubeslice-cli/topology-configuration.mdx (hunk 4 above)
Comment (uma-kt) [reply versioned_docs/version-1.0.0/install-kubeslice/kubeslice-cli/topology-configuration.mdx:99]: - same as above

File: versioned_docs/version-1.0.0/install-kubeslice/kubeslice-cli/topology-configuration.mdx (hunk 5 above)
Comment (uma-kt) [reply versioned_docs/version-1.0.0/install-kubeslice/kubeslice-cli/topology-configuration.mdx:108]: - same as above

**Review by veenamj-avesha — COMMENTED** [review]
I have left questions on the UI chart and enterprise demo which don't hold good to the OSS edition AFAIK. 

Can you address them?

File: versioned_docs/version-1.1.0/install-kubeslice/kubeslice-cli/topology-configuration.mdx [hunk 9]
```diff
@@ -146,4 +176,21 @@ configuration:
       chart_name: #{The name of the Worker Chart}
       version: #{The version of the chart to use. Leave blank for latest version}
       values: #(Values to be passed as --set arguments to helm install)
+    
+    # The ui_chart, prometheus_chart, and image_pull_secret parameters are required for enterprise chart only
+    ui_chart:
```
[the remark this answers is not in the document]
Comment (uma-kt) [reply versioned_docs/version-1.1.0/install-kubeslice/kubeslice-cli/topology-configuration.mdx:181]: - same as above

File: versioned_docs/version-1.0.0/install-kubeslice/kubeslice-cli/topology-configuration.mdx (hunk 4 above)
Comment (veenamj-avesha) [reply versioned_docs/version-1.0.0/install-kubeslice/kubeslice-cli/topology-configuration.mdx:99]: Does it work with minimal and full demo? Does it only work for enterprise demo? We need to let the user know that right?

File: versioned_docs/version-1.0.0/install-kubeslice/kubeslice-cli/topology-configuration.mdx (hunk 5 above)
Comment (veenamj-avesha) [reply versioned_docs/version-1.0.0/install-kubeslice/kubeslice-cli/topology-configuration.mdx:108]: We need to let the user know that it is only required with `enterprise-demo` in our docs (even if it's not there in the tool help description), How will the user know how to get the image pull secrets? Does the tool show up the registration form? 

It's also good to have this https://docs.avesha.io/documentation/enterprise/1.3.0/playground/playground-ent-kubeslice-demo-kind#enterprise-artifact-registration description added as a cross reference. 

cc @narmidm. 

I discussed this with Imran and he told me that it's good to let the user know about the Enterprise demo. 

We are retaining it but we should let the user know that UI charts, image pull charts and/or Prometheus charts are required only with the enterprise-demo option...

File: versioned_docs/version-1.0.0/install-kubeslice/kubeslice-cli/topology-configuration.mdx [hunk 10]
```diff
@@ -146,4 +176,21 @@ configuration:
       chart_name: #{The name of the Worker Chart}
       version: #{The version of the chart to use. Leave blank for latest version}
       values: #(Values to be passed as --set arguments to helm install)
+    
+    # The ui_chart, prometheus_chart, and image_pull_secret parameters are required for enterprise chart only
```
Comment (veenamj-avesha) [inline versioned_docs/version-1.0.0/install-kubeslice/kubeslice-cli/topology-configuration.mdx:180]: Change `enterprise chart ` only to enterprise-demo (enterprise chart)  to be more clear.

File: versioned_docs/version-1.0.0/install-kubeslice/kubeslice-cli/topology-configuration.mdx [hunk 11]
```diff
@@ -55,9 +55,11 @@ The following table describes the helm chart configuration parameters.
 | [cert_manager_chart](#certificate-manager-chart-configuration-parameters) | Object         | The cert-manager chart details.                                                                | Mandatory |
 | [controller_chart](#controller-chart-configuration-parameters)            | Object         | The Kubeslice Controller chart details.                                                        | Mandatory |
 | [worker_chart](#worker-chart-configuration-parameters)                    | Object         | The KubeSlice Worker chart details.                                                            | Mandatory |
+| [ui_chart](#ui-chart-configuration-parameters)                            | Object         | The KubeSlice Manager chart details. Required for Enterprise charts only.                      | Optional  |
```
Comment (veenamj-avesha) [inline versioned_docs/version-1.0.0/install-kubeslice/kubeslice-cli/topology-configuration.mdx:58]: Change `Required for Enterprise charts only` to `This object is only required for enterprise charts.` 

Full sentence is better, and emphasize on what's only required.

File: versioned_docs/version-1.0.0/install-kubeslice/kubeslice-cli/topology-configuration.mdx [hunk 12]
```diff
@@ -55,9 +55,11 @@ The following table describes the helm chart configuration parameters.
...
+| [prometheus_chart](#prometheus-chart-configuration-parameters)            | Object         | The Prometheus chart details. Required for Enterprise charts only.                             | Optional  |
 | helm_username                                                             | AlphaNumeric   | The helm username if the repo is private.                                                      | Optional  |
 | helm_password                                                             | AlphaNumeric   | The helm password if the repo is private.                                                      | Optional  |
-| [image_pull_secret](#image-pull-configuration-parameters)                 | Object         | The image pull secrets to pull the Kubernetes dashboard image. Required for Enterprise charts. | Mandatory |
+| [image_pull_secret](#image-pull-configuration-parameters)                 | Object         | The image pull secrets to pull the Kubernetes dashboard image. Required for Enterprise charts only. | Mandatory |
```
Comment (veenamj-avesha) [inline versioned_docs/version-1.0.0/install-kubeslice/kubeslice-cli/topology-configuration.mdx:62]: Change `Required for Enterprise charts only` to `This object is only required for enterprise charts.` 

Full sentence is better, and emphasize on what's only required.

File: versioned_docs/version-1.0.0/install-kubeslice/kubeslice-cli/topology-configuration.mdx [hunk 13]
```diff
@@ -85,6 +87,34 @@ The following table describes the worker chart configuration parameters used to
 | version    | Numeric        | The version of the chart. Leave this value blank for the latest version.         | Optional  |
 | values     | String         | The additional value that can be passed as --set arguments for **helm install**. | Optional  |
 
+## UI Chart Configuration Parameters
+The following table describes the UI chart configuration parameters used to create topology YAML.
```
Comment (veenamj-avesha) [inline versioned_docs/version-1.0.0/install-kubeslice/kubeslice-cli/topology-configuration.mdx:91]: Add `These parameters are only required when you use the `enterprise-demo` option. `

File: versioned_docs/version-1.0.0/install-kubeslice/kubeslice-cli/topology-configuration.mdx [hunk 14]
```diff
@@ -85,6 +87,34 @@ The following table describes the worker chart configuration parameters used to
...
+
+| Parameter  | Parameter Type | Description                                                                      | Required  |
+| ---------- | -------------- | -------------------------------------------------------------------------------- | --------- |
+| chart_name | String         | The name of the UI or enterprise chart.                                          | Mandatory |
+| version    | Numeric        | The version of the chart. Leave it blank for the latest version.                 | Optional  |
+| values     | String         | The additional value that can be passed as --set arguments for **helm install**. | Optional  |
+
+## Prometheus Chart Configuration parameters
+The following table describes the Prometheus chart configuration parameters used to create topology YAML.
```
Comment (veenamj-avesha) [inline versioned_docs/version-1.0.0/install-kubeslice/kubeslice-cli/topology-configuration.mdx:100]: Add this as a second sentence after topology YAML. 

`These parameters are only required when you use the enterprise-demo option. `

File: versioned_docs/version-1.0.0/install-kubeslice/kubeslice-cli/topology-configuration.mdx [hunk 15]
```diff
@@ -85,6 +87,34 @@ The following table describes the worker chart configuration parameters used to
...
+
+| Parameter  | Parameter Type | Description                                                                      | Required  |
+| ---------- | -------------- | -------------------------------------------------------------------------------- | --------- |
+| chart_name | String         | The name of the Prometheus chart.                                                | Mandatory |
+| version    | Numeric        | The version of the chart. Leave it blank for the latest version.                 | Optional  |
+| values     | String         | The additional value that can be passed as --set arguments for **helm install**. | Optional  |
+
+## Image Pull Secret Configuration Parameters
+The following table describes the image pull secret configuration parameters used to create topology YAML.
```
Comment (veenamj-avesha) [inline versioned_docs/version-1.0.0/install-kubeslice/kubeslice-cli/topology-configuration.mdx:109]: Add this as a second sentence after topology YAML. 

`These parameters are only required when you use the enterprise-demo option. `

File: versioned_docs/version-1.0.0/install-kubeslice/kubeslice-cli/topology-configuration.mdx (hunk 14 above)
Comment (uma-kt) [reply versioned_docs/version-1.0.0/install-kubeslice/kubeslice-cli/topology-configuration.mdx:100]: Added the below
These parameters are required for the `enterprise-demo` option only.

File: versioned_docs/version-1.0.0/install-kubeslice/kubeslice-cli/topology-configuration.mdx (hunk 13 above)
Comment (uma-kt) [reply versioned_docs/version-1.0.0/install-kubeslice/kubeslice-cli/topology-configuration.mdx:91]: These parameters are required for the `enterprise-demo` option only.

File: versioned_docs/version-1.1.0/install-kubeslice/kubeslice-cli/topology-configuration.mdx [hunk 16]
```diff
@@ -146,4 +179,21 @@ configuration:
       chart_name: #{The name of the Worker Chart}
       version: #{The version of the chart to use. Leave blank for latest version}
       values: #(Values to be passed as --set arguments to helm install)
+    
+    # The ui_chart, prometheus_chart, and image_pull_secret parameters are required for the enterprise-demo option (enterprise-chart) only
```
Comment (veenamj-avesha) [inline versioned_docs/version-1.1.0/install-kubeslice/kubeslice-cli/topology-configuration.mdx:183]: Add a full stop after `only` to complete the sentence. This comment is valid for the 1.0.0 topology configuration file also.

Comment (uma-kt) [reply versioned_docs/version-1.1.0/install-kubeslice/kubeslice-cli/topology-configuration.mdx]: done

**Pull request merged**